xunit.runner.worker/Connection.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace xunit.runner.worker
+{
+    internal abstract class Connection : IDisposable
+    {
+        private bool _closed;
+
+        internal abstract Stream Stream { get; }
+
+        internal abstract void WaitForClientConnect();
+        internal abstract void WaitForClientDone();
+
+        protected virtual void DisposeCore()
+        {
+
+        }
+
+        private void Dispose()
```

Comment:
This is an old habit of mine for personal projects.  I typically break interface implementation into a) explicit implementation and b) the non-explicit method which does the work.  I'm not wedded to it if you'd prefer a different way.